src/Symfony/Component/Console/Application.php

```diff
@@ -745,10 +726,15 @@ public function getTerminalDimensions()
      * @param int $height The height
      *
      * @return Application The current application
+     *
+     * @deprecated since version 3.2, to be removed in 4.0. Set the COLUMNS and LINES env vars instead.
      */
     public function setTerminalDimensions($width, $height)
     {
-        $this->terminalDimensions = array($width, $height);
+        @trigger_error(sprintf('%s is deprecated as of 3.2 and will be removed in 4.0. Set the COLUMNS and LINES env vars instead.', __METHOD__, ArgumentResolverInterface::class), E_USER_DEPRECATED);
+
+        putenv('COLUMNS='.$width);
+        putenv('LINES='.$height);
```

Comment:
this does not work in case the dimensions were already retrieved previously, because of the Terminal internal cache